src/EventHub/EventHub/Cmdlets/Clusters/SetAzureEventHubCluster.cs

```diff
@@ -0,0 +1,111 @@
+    public class SetAzureEventHubCluster : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Cluster Name")]
+        [Parameter(Mandatory = true, ParameterSetName = ClusterResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Cluster Name")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Location of Cluster")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Cluster Capacity (CU), curerntrly, allowed value = 1")]
+        [ValidateNotNullOrEmpty]
+        public int? Capacity { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ClusterInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Cluster Name")]
+        [ValidateNotNullOrEmpty]
+        public PSEventHubClusterAttributes InputObject { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ClusterResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Resource ID of Cluster")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Hashtables which represents resource Tags for Clusters")]
+        [Parameter(Mandatory = false, ParameterSetName = ClusterResourceIdParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Hashtables which represents resource Tags for Clusters")]
+        public Hashtable Tag { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            PSEventHubClusterAttributes cluster = new PSEventHubClusterAttributes();
+
+            if (ParameterSetName.Equals(ClusterInputObjectParameterSet))
+            {
+                cluster = InputObject;
+                cluster.Name = Name;
+                cluster.Location = InputObject.Location;
+            }
+
+            if (ParameterSetName.Equals(ClusterPropertiesParameterSet))
+            {
+                if (Tag != null)
```

Comment:
Please use `this.IsParameterBound(c => c.Tag)` to check the valid of parameter.

Reply:
implemented